compiler/rustc_builtin_macros/src/test.rs

```diff
@@ -523,72 +526,64 @@ fn test_type(cx: &ExtCtxt<'_>) -> TestType {
     }
 }
 
-fn has_test_signature(cx: &ExtCtxt<'_>, i: &ast::Item) -> bool {
+fn check_test_signature(
+    cx: &ExtCtxt<'_>,
+    i: &ast::Item,
```

Comment:
The only uses of `i` here and below appear to be the span. To avoid confusion, could you only pass the span instead of the entire item?

Reply:
The item is also used in
```rust
let has_should_panic_attr = attr::contains_name(&i.attrs, sym::should_panic);
```

I tried moving that out too, but IMO it made the function arguments too cluttered.

Reply:
ah, fine